### PR TITLE
Add toggle to inject the tenant ID to generated metrics

### PR DESCRIPTION
**Pull request by kvrhdn**
**What this PR does**:

When a `inject_tenant_id_as` is set, the Tempo tenant ID is added to all metrics generated by the metrics-generator.
E.g. if set to `__tempo_tenant`, metrics generated for tenant `foo` will all have the label `__tempo_tenant="foo"`.

```yaml
metrics_generator:
  registry:
    inject_tenant_id_as: "__tempo_tenant"
```

I made the config a string so users can pick a value that doesn't conflict with other labels.

This should make Tempo metrics-generator more useful with a single-tenant metrics backend.

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/tempo/issues/3630

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/configuration/_index.md [hunk 1]
```diff
@@ -379,6 +379,9 @@ metrics_generator:
         # A list of labels that will be added to all generated metrics.
         [external_labels: <map>]
 
+        # If set, the tenant ID will added as label with the given label name to all generated metrics.
+        [inject_tenant_id_as: <string>]
```
Comment (kvrhdn) [inline docs/sources/tempo/configuration/_index.md:383]: I'll be honest, I didn't really find a nice and concise way to describe this setting. Suggestion are welcome 😅

Also feedback about the name of the setting itself is welcome. It's a bit verbose now imo.

Comment (mdisibio) [reply docs/sources/tempo/configuration/_index.md:383]: I like it 👍

**Review by mdisibio — APPROVED** [review]
Nice feature and lgtm.

File: docs/sources/tempo/configuration/_index.md (hunk 1 above)
Comment (knylander-grafana) [reply docs/sources/tempo/configuration/_index.md:383]: Even if it's long, it's clear what it does.

**Review by knylander-grafana — COMMENTED** [review]
Approving for text and doc updates.

**Comment by kvrhdn** [comment]
Squashing so backport action can cherry pick the commit

**Pull request merged**